src/cpp/CMakeLists.txt

```
@@ -76,7 +76,29 @@ target_include_directories(${TARGET_NAME_OBJ}
            "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>"
     PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src")
 
-target_include_directories(${TARGET_NAME_OBJ} SYSTEM PRIVATE "${safetensors.h_SOURCE_DIR}")
+if(ENABLE_GGUF)
+    message(STATUS "Downloading gguflib")
+    FetchContent_Declare(
+      gguflib
+      GIT_REPOSITORY https://github.com/antirez/gguf-tools/
+      GIT_TAG af7d88d808a7608a33723fba067036202910acb3)
+    FetchContent_MakeAvailable(gguflib)
+    target_include_directories(${TARGET_NAME_OBJ}
+                               PRIVATE "${gguflib_SOURCE_DIR}")
+    target_include_directories(${TARGET_NAME_OBJ}
+                              PRIVATE "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/gguf_utils>")
```

```suggestion
                              PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/src/gguf_utils")
```
`BUILD_INTERFACE` makes sense only for `INTERFACE` / `PUBLIC` key words, because `PRIVATE` headers are not exposed anyway